sdk/js/src/api/stream/stream.js

```diff
@@ -96,6 +97,10 @@ export default async function (payload, url) {
           notify(listeners[EVENTS.ERROR], error)
           listeners = null
         })
+    }).catch(function (error) {
```

Comment:
If we don't add this, the error will be swallowed, because it occurs in a promise that runs after the subscription object has been returned.

What other implementation do you suggest?

Reply:
I've tried several implementations here:
- Awaiting the initial request ⟶ makes us lose the start event (which I think is not an issue, is also similar to how [`nes`](https://github.com/hapijs/nes#client-1) works)
- Exposing an `open()` function to the returned subscription object, to be able to open the stream after the handlers have been set ⟶ messes up our SDK API
- Using current implementation with added catch block ⟶ makes `error` event ambiguous (event error, or stream start error)

Reply:
1. https://gist.github.com/bafonins/1a8d3f7840652fb1e5756c69d3a14df6 - something like this, can add some `async/await` to make it prettier. note, it doesnt really matter where you call the `open()` method (before/after adding listeners). dont really this approach.
2. https://gist.github.com/bafonins/2e60e9f28b5fd13b7c9aecf67348e74f - `resolve/reject` based on `fetch` result and propagate errors to the user 
3. add `.catch` to the `fetch` request and notify listeners on `events.error`, but then we have to clearly separate errors caused by reading chunked data and errors caused by failed `fetch`.

Reply:
Took inspiration from your approaches to implement it in a way that will allow us to await the request (like my first bullet point) without losing the `start` event.

This allows us to keep the original usage of the stream in the event logic.